### PR TITLE
catboost: adding option to build R library

### PR DESCRIPTION
**Pull request by nviets**
Adding option to build to build the R library. See included comments for an example of usage.

@natsukium, @jbedo 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/catboost/default.nix [hunk 1]
```diff
@@ -79,24 +97,45 @@ stdenv.mkDerivation (finalAttrs: {
   cmakeFlags = [
     "-DCMAKE_BINARY_DIR=$out"
     "-DCMAKE_POSITION_INDEPENDENT_CODE=on"
-    "-DCATBOOST_COMPONENTS=app;libs${lib.optionalString pythonSupport ";python-package"}"
+    "-DCATBOOST_COMPONENTS=app;libs${lib.optionalString pythonSupport ";python-package"}${lib.optionalString rLibrary ";R-package"}"
   ] ++ lib.optionals cudaSupport [
-    "-DHAVE_CUDA=on"
+    "-DHAVE_CUDA=yes"
```
Comment (natsukium) [inline pkgs/development/libraries/catboost/default.nix:102]: I don't mind changing it, but I want to keep it consistent.

File: pkgs/development/libraries/catboost/default.nix [hunk 2]
```diff
@@ -79,24 +97,45 @@ stdenv.mkDerivation (finalAttrs: {
...
+    cp -r $src/catboost/R-package/* catboost/R-package
+  '';
+
   installPhase = ''
     runHook preInstall
-
+  '' + lib.optionalString (!rLibrary) ''
     mkdir $dev
     cp -r catboost $dev
     install -Dm555 catboost/app/catboost -t $out/bin
```
Comment (natsukium) [inline pkgs/development/libraries/catboost/default.nix:120]:
```suggestion
    mkdir $dev
    cp -r catboost $dev
  '' + lib.optionalString (!rLibrary) ''
    install -Dm555 catboost/app/catboost -t $out/bin
```
I guess the python library is not built with `rLibrary=true; pythonSupport=true;`.

I think the installPhase will need to be reconstructed later.

File: pkgs/development/libraries/catboost/default.nix (hunk 1 above)
Comment (nviets) [reply pkgs/development/libraries/catboost/default.nix:102]: [Catboost's latest documentation](https://catboost.ai/en/docs/installation/build-native-artifacts) indicates yes|no. Is there any reason to go with on|off instead?

**Comment by nviets** [comment]
Thanks for your suggestions, @natsukium. I have updated the PR.

**Comment by nviets** [comment]
@natsukium - gentle reminder. thanks!

File: pkgs/development/libraries/catboost/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/development/libraries/catboost/default.nix:102]: There is no particular reason for on/off, just personal preference. (I really wanted to use uppercase, but for some reason, it's lowercase.)

**Review by natsukium — CHANGES_REQUESTED** [review]
Please change the commit message to `catboost: adding option to build R library`.

**Comment by natsukium** [comment]
It works fine on my x86_64-linux with/without cuda & aarch64-darwin machines with the following command.
```
nix-shell -p 'rWrapper.override { packages = [ (catboost.override { rLibrary = true; }) ]; }'
```

**Comment by cole-h** [comment]
I'm locking and closing this pull request because you accidentally requested reviews from a lot of people. Please open a new PR. To avoid this problem in the future, you can mark the PR as a draft before you rebase. While the PR is a draft, you will be able to see the list of maintainers that will be requested for review, before they all receive a notification.

**Pull request closed without merging**